### DISCUSSION TITLE
ddns-updater (.jason file syntax)

### DISCUSSION THREAD

**Discussion by kattivius** [qdm12/ddns-updater#749] [General]
Greetings,
I have my .jason file configured to use with DDNS-UPDATER (_docker container_). it all works fine but I am looking for a nicer and shorter syntax. 
**NOTE**
if you actually define a subdomain in the form of _sub.domain.com_ you end up having _sub.domain.com.domain.com_ at porkbun.
If you only define _sub1_ (no domain, only prefix), it works. (sub1.domain.com)
Here is how I have it and what I tried already: (note is now used to update with porkbun)
Now works likes this:

```json
{
  "settings": [
    {
      "provider": "porkbun",
      "domain": "domain.org",
      "host": "Sub1",
      "api_key": "mykey",
      "secret_api_key": "mysecret",
      "ip_version": "ipv4",
      "ipv6_suffix": ""
    },
    {
      "provider": "porkbun",
      "domain": "domain.org",
      "host": "Sub2",
      "api_key": "mykey",
      "secret_api_key": "mysecret",
      "ip_version": "ipv4",
      "ipv6_suffix": ""
    },
    {
      "provider": "porkbun",
      "domain": "domain.org",
      "host": "Sub3",
      "api_key": "mykey",
      "secret_api_key": "mysecret",
      "ip_version": "ipv4",
      "ipv6_suffix": ""
    }
  ]
}
```
 
The following is what I tried but did not work:

```json
{
  "settings": [
    {
      "provider": "porkbun",
      "domain": "domain.com",
      "host": [
        "sub1",
        "sub2",
        "sub3"
      ],
      "api_key": "key",
      "secret_api_key": "secret",
      "ip_version": "ipv4",
      "ipv6_suffix": ""
    }
  ]
}
```

 
also tried this:

```json 
{
  "settings": [
    {
      "provider": "porkbun",
      "domain": "domain.com",
      "host": ["sub1", "sub2"],
      "api_key": "sk1_7d119e3f656b00ae042980302e1425a04163c476efec1833q3cb0w54fc6f5022",
      "secret_api_key": "pk1_5299b57125c8f3cdf347d2fe0e713311ee3a1e11f11a14942b26472593e35368",
      "ip_version": "ipv4",
      "ipv6_suffix": ""
    }
  ]
}
```
 
This last one only takes the first subsomain in consideration

Do you know how to have this configuration that actually works using the multiple "Host" on the same line rather than repeating the all section for each host?

**Comment by qdm12** [comment]
> I am looking for a nicer and shorter syntax.

Use `"host": "sub1,sub2"`, it's written in the readme 😉 No worry, I'm not judging you for skipping over this, the readme isn't that small!

> you can specify multiple hosts for the same domain using a comma separated list. For example with "host": "@,subdomain1,subdomain2",.

Regarding the `host` naming, it's a bit of bad naming, it should be `owner` technically. See in [this issue comment](https://github.com/qdm12/ddns-updater/issues/625#issuecomment-1970810326) the point number **1.**

For further simplications, see in that same issue comment the point number **2** which will allow the user to just specify a `"domain"` field with everything in there. For example `abc.domain.com`, `*.domain.com`, `domain.com` etc. in a still retro-compatible way.
